src/Dhall/Import/Types.hs

```diff
@@ -17,7 +17,11 @@ import Dhall.TypeCheck (X)
 
 -- | State threaded throughout the import process
 data Status = Status
-    { _stack   :: [Import]
+    { _root :: FilePath
```

Comment:
You don't need to add a `_root` field.  Instead, you can make `_stack` a `NonEmpty` list of `Import`s and set the first one to the directory with a file named `"."`